rpki-validator/src/main/java/net/ripe/rpki/validator3/background/BackgroundJobs.java

```diff
@@ -51,72 +46,71 @@
 import java.time.Clock;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.Date;
-import java.util.Map;
 import java.util.SortedMap;
 import java.util.TreeMap;
-
-import static org.quartz.DateBuilder.IntervalUnit.MINUTE;
-import static org.quartz.DateBuilder.IntervalUnit.SECOND;
-import static org.quartz.DateBuilder.futureDate;
-import static org.quartz.JobBuilder.newJob;
-import static org.quartz.SimpleScheduleBuilder.simpleSchedule;
-import static org.quartz.TriggerBuilder.newTrigger;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 
 @Profile("!test")
 @Component
 @Slf4j
 public class BackgroundJobs extends JobListenerSupport {
 
-    private final Scheduler scheduler;
-
-    @Autowired
-    public BackgroundJobs(Scheduler scheduler) throws SchedulerException {
+    private final ScheduledExecutorService scheduledExecutorService;
 
-        this.scheduler = scheduler;
+    private final RpkiObjectCleanupService rpkiObjectCleanupService;
+    private final RpkiRepositoryCleanupService rpkiRepositoryCleanupService;
+    private final ValidationRunCleanupService validationRunCleanupService;
+    private final RpkiRepositoryValidationService rpkiRepositoryValidationService;
+    private final BgpPreviewService bgpPreviewService;
 
-        scheduler.getListenerManager().addJobListener(this);
+    @Autowired
+    public BackgroundJobs(RpkiObjectCleanupService rpkiObjectCleanupService,
+                          RpkiRepositoryCleanupService rpkiRepositoryCleanupService,
+                          ValidationRunCleanupService validationRunCleanupService,
+                          RpkiRepositoryValidationService rpkiRepositoryValidationService,
+                          BgpPreviewService bgpPreviewService) {
+        this.scheduledExecutorService = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors());
```

Comment:
Quartz uses a fixed pool of 10 threads by default. The `newScheduledThreadPool` has no maximum by default. We could just use `0` as the core pool size and let the thread pool create threads as needed, there should be little overhead.

Reply:
Indeed!
From what I can see there's not way to create ScheduledThreadPoolExecutor with a cap on the thread pool size. No idea why actually, to always have a thread to submit to? Will make it 0 then.